File: 2026 06월/6월 1주차/kizizip/02/[PGS - Lv1] 제일 작은 수 제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        if (arr.length == 1) {
            return new int[] {-1};
        }

        int min = arr[0];
        int minIdx = 0;

        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < min) {
                min = arr[i];
                minIdx = i;
            }
        }

        int[] answer = new int[arr.length - 1];
        int idx = 0;

        for (int i = 0; i < arr.length; i++) {
            if (i != minIdx) {
                answer[idx++] = arr[i];
            }
        }

        return answer;
    }
}

File: JeongEon8/02/[PGS - Lv1] 제일 작은 수 제거하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    
    int smallest = 0;
    int size = arr.size();
    
    for(int i = 0; i < size; i++){
        if(arr[smallest] > arr[i]){
            smallest = i;
        }
    }
    
    arr.erase(arr.begin() + smallest);
    if(arr.empty()){
        arr.push_back(-1);
    }
    
    answer = arr;
    
    return answer;
}
